drivers/wdf/wdfldr/common.c

```diff
@@ -0,0 +1,826 @@
+/*
+ * PROJECT:     ReactOS WdfLdr driver
+ * LICENSE:     GPL-2.0-or-later (https://spdx.org/licenses/GPL-2.0-or-later)
+ * PURPOSE:     WdfLdr driver - common functions
+ * COPYRIGHT:   Copyright 2019 mrmks04 (mrmks04@yandex.ru)
+ */
+
+
+#include "common.h"
+#include "ntddk_ex.h"
+
+#include <ntintsafe.h>
+#include <ntstrsafe.h>
+
+
+NTSTATUS
+NTAPI
+AuxKlibQueryModuleInformation(
+	IN PULONG InformationLenght,
+	IN ULONG SizePerModule,
+	IN OUT PRTL_MODULE_EXTENDED_INFO ModuleInfo
+);
+
+
+#ifdef ALLOC_PRAGMA
+#pragma alloc_text (PAGE, AuxKlibQueryModuleInformation)
+#endif
+
+
+VOID
+NTAPI
+GetNameFromPath(
+	IN PUNICODE_STRING Path,
+	OUT PUNICODE_STRING Name
+)
+{
+	PWCHAR nextSym;
+	PWCHAR currSym;
+
+	if (Path->Length >= 2) 
+	{
+		Name->Buffer = (wchar_t*)((char*)Path->Buffer + Path->Length - 2);
+		Name->Length = 2;
+
+		for (nextSym = Name->Buffer; ; Name->Buffer = nextSym) 
+		{
+			if (nextSym < Path->Buffer) 
+			{
+				Name->Length -= 2;
+				++Name->Buffer;
+				goto end;
+			}
+			currSym = Name->Buffer;
+
+			if (*currSym == '\\')
+			{
+				break;
+			}
+			nextSym = currSym - 1;
+			Name->Length += 2;
+		}
+
+		++Name->Buffer;
+		Name->Length -= 2;
+
+		if (Name->Length == 2) 
+		{
+			Name->Buffer = NULL;
+			Name->Length = 0;
+		}
+	end:
+		Name->MaximumLength = Name->Length;
+	}
+	else 
+	{
+		Name->Length = 0;
+		Name->Buffer = NULL;
+	}
+}
+
+
+NTSTATUS
+NTAPI
+GetImageName(
+	IN PUNICODE_STRING DriverServiceName,
+	IN ULONG Tag,
+	IN PUNICODE_STRING ImageName
+)
+{
+	NTSTATUS status;
+	OBJECT_ATTRIBUTES ObjectAttributes;
+	UNICODE_STRING name;
+	UNICODE_STRING path;
+	PKEY_VALUE_PARTIAL_INFORMATION pKeyValPartial;
+	HANDLE KeyHandle;
+	UNICODE_STRING ValueName;
+
+	ImageName->Length = 0;
+	ImageName->Buffer = NULL;
+	name.Length = 0;
+	name.Buffer = NULL;
+	KeyHandle = NULL;
+	pKeyValPartial = NULL;
+	RtlInitUnicodeString(&ValueName, L"ImagePath");
+
+	ObjectAttributes.ObjectName = DriverServiceName;
+	ObjectAttributes.Length = 24;
+	ObjectAttributes.RootDirectory = 0;
+	ObjectAttributes.Attributes = OBJ_KERNEL_HANDLE | OBJ_CASE_INSENSITIVE;//576; OBJ_OPENIF;
+	ObjectAttributes.SecurityDescriptor = 0;
+	ObjectAttributes.SecurityQualityOfService = 0;
```

Comment:
Can you use the appropriate macro InitializeObjectAttributes() and avoid magic value?